spec/core/data_structures.md

```diff
@@ -4,195 +4,243 @@ Here we describe the data structures in the Tendermint blockchain and the rules
 
 The Tendermint blockchains consists of a short list of basic data types:
 
-- `Block`
-- `Header`
-- `Version`
-- `BlockID`
-- `Time`
-- `Data` (for transactions)
-- `Commit` and `Vote`
-- `EvidenceData` and `Evidence`
+- [`Block`](#block)
+- [`Header`](#header)
+- [`Version`](#version)
+- [`BlockID`](#blockid)
+- [`Time`](#time)
+- [`Data` (for transactions)](#data)
+- [`Commit`](#commit)
+- [`Vote`](#vote)
+- [`EvidenceData`](#evidence_data)
+- [`Evidence`](#evidence)
 
 ## Block
 
 A block consists of a header, transactions, votes (the commit),
 and a list of evidence of malfeasance (ie. signing conflicting votes).
 
-```go
-type Block struct {
-    Header      Header
-    Txs         Data
-    Evidence    EvidenceData
-    LastCommit  Commit
-}
-```
+| Name       | Type                           | Description                                                                                                                                                                          | Validation                                                                                       |
+|------------|--------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|
+| Header     | [Header](#header)              | Header corresponding to the block. This field contains information used throughout consensus and other areas of the protocol. To find out what it contains, visit [header] (#header) | Must adhere to the validation rules of [header](#header)                                         |
+| Data       | [Data](#data)                  | Data contains a list of transactions. The contents of the transaction is unknown to Tendermint.                                                                                      | This field can be empty or populated, but the data is unknown to Tendermint                      |
```

Comment:
You sort of mention that data is unknown to Tendermint twice. On another note we could add that validation of txs is reliant on the application i.e `CheckTx()`

Reply:
This is true, but the block validation does not rely on checkTx. I can add a minor reference